### ISSUE TITLE
Connections

### ISSUE THREAD

**Issue by karlabbott** [vector-im/element-meta#137]
**Is your feature request related to a problem? Please describe.**
The more people and the more rooms that I talk to on Element, the more difficult it becomes to organize it all and find the most important conversations to me. Did I favorite that person? Did I favorite that room? Where is that notification in this space coming from? Scroll time.

**Describe the solution you'd like**
I'd like to describe a feature that I call "Connections" (working name, but apropos) that would replace Favorites/People/Rooms and provide the same functionality while better ordering the list of possible "connections". 

How "Connections" works is as follows. You have a section under Home (or any space really) that looks like this:

~~~
------------------------------------------------------------------------------
Connections                                    Hide Rooms / Favorites

Friendly Person (1)
Work Colleague (1)
Matrix HQ
Beyond Chat
Coffee
------------------------------------------------------------------------------
~~~

- This Connections listing is organized by activity
- 1:1 messages **always** bubble up to the top
- Rooms bubble up below 1:1 messages based on activity
- Rooms should then be naturally ordered by notification preferences. A room with a keyword or mention highlight should take precedence over a new message or a muted room.

Further, in the above interface, if you click "Hide Rooms" you would see:

~~~
------------------------------------------------------------------------------
Connections                                    Show Rooms / Favorites

Friendly Person
Work Colleague
------------------------------------------------------------------------------
~~~

As you can see, we no longer see the rooms and we just see direct 1:1 chats.

Further, if we click Favorites, we can filter the list down to rooms and contacts that have been favorited. Once the favorites filter has been selected, the text for Favorites should probably turn into "Show all" or something on those lines.

It's also possible that if none of your 1:1 connections has said anything recently, you could end up with this:

~~~
------------------------------------------------------------------------------
Connections                                    Hide Rooms / Favorites

Matrix HQ (1)
Beyond Chat (1)
Friendly Person 
Work Colleague
Coffee
------------------------------------------------------------------------------
~~~

But the moment that Friendly Person sends you a message, you now see:

~~~
------------------------------------------------------------------------------
Connections                                    Hide Rooms / Favorites

Friendly Person (1)
Matrix HQ (1)
Beyond Chat (1)
Work Colleague
Coffee
------------------------------------------------------------------------------
~~~

This ordering persists **even** if Matrix HQ has had a more recent update to its event graph than Friendly Person.

What happens if you can't find your 1:1 rooms because no one has said anything recently? Just click Hide Rooms and you'll be left with who's currently in the People list. (see above for an example of clicking "Hide Rooms".)

I do believe that implementing this concept or something like it would be a cleaner presentation of the Favorites/People/Rooms ordering that we have today.

**Describe alternatives you've considered**
You can get pretty close to this if you favorite all the people and rooms that you care about, minimize your people and rooms lists, and then sort your favorites by activity. This works pretty well, but requires a decent bit of setup by the user and still does not prioritize 1:1 rooms.

**Additional context**
To some degree, the article https://blogs.gnome.org/tbernard/2018/05/16/banquets-and-barbecues/ helped inspire my thought process behind this. I don't think that splitting apps out or trying to oversimplify things is really the answer here. That said, by simply making some interface tweaks, I believe that what I've proposed would likely be cleaner, more intuitive and give us more mileage out of this panel in Element.

**Comment by karlabbott** [comment]
I don't like having to favorite everybody to achieve this type of list, but with metaspaces, this is manageable. I'm wondering if an idea like the above would be a better version of "Home" for mobile, in particular, iOS? One vertical list as opposed to 3 horizontally scrolling lists...

**Comment by daniellekirkwood** [comment]
This was discussed in Office Hours today and @kittykat feels like feedback that's relevant as y'all start on the journey of Mobile IA

**Comment by ilu33** [comment]
This sounds really complicated so I'm not sure about it - what I'd like to have:
An option to sort everything according to activity only. Distinguishing in between DMs and rooms makes absolutely no sense to me (the third person joining is often just me with my second account). Since we now have private spaces (thanks for that!) I can sort how I want (family, club, job, ...) but I still have to manually favorite-ize all my rooms and DMs which sucks especially on mobile.

**Comment by kittykat** [comment]
> Distinguishing in between DMs and rooms makes absolutely no sense to me (the third person joining is often just me with my second account). Since we now have private spaces (thanks for that!) I can sort how I want (family, club, job, ...) but I still have to manually favorite-ize all my rooms and DMs which sucks especially on mobile.

In case you didn't know, this may work as a workaround for now: you can mark rooms as DMs in the room setting on iOS or with `/converttodm` command on web.

**Comment by ilu33** [comment]
Yes, I know - but wouldn't that change the nature of the room for everybody else too? Other people might not like that change.

Having a way to get everything sorted (by latest comment) into one consolidated list per space would still be good.
